txtorcon/socks.py

```diff
@@ -203,9 +200,9 @@ def _parse_request_reply(self):
             return
 
         if reply != self.SUCCEEDED:
-            try:
-                self.reply_error(_socks_reply_code_to_string[reply])
-            except KeyError:
+            if reply in SocksErrorFactory.socks_errors:
+                self.reply_error(reply)
+            else:
                 self.reply_error("Unknown SOCKS reply code {}".format(reply))
```

Comment:
With the suggested factory below I think this can just be `_create_socks_error(reply)` with no checks, but if not I prefer the `try/except KeyError` over "check, then access".